src/mqt/predictor/rl/actions.py

```diff
@@ -0,0 +1,445 @@
+    CliffordSimp,
+    FullPeepholeOptimise,
+    PeepholeOptimise2Q,
+    RemoveRedundancies,
+    RoutingPass,
+)
+from qiskit.circuit import StandardEquivalenceLibrary
+from qiskit.circuit.library import XGate, ZGate
+from qiskit.passmanager import ConditionalController
+from qiskit.transpiler import (
+    CouplingMap,
+)
+from qiskit.transpiler.passes import (
+    ApplyLayout,
+    BasicSwap,
+    BasisTranslator,
+    Collect2qBlocks,
+    CommutativeCancellation,
+    CommutativeInverseCancellation,
+    ConsolidateBlocks,
+    DenseLayout,
+    Depth,
+    EnlargeWithAncilla,
+    FixedPoint,
+    FullAncillaAllocation,
+    GatesInBasis,
+    InverseCancellation,
+    MinimumPoint,
+    Optimize1qGatesDecomposition,
+    OptimizeCliffords,
+    RemoveDiagonalGatesBeforeMeasure,
+    SabreLayout,
+    Size,
+    TrivialLayout,
+    UnitarySynthesis,
+    VF2Layout,
+    VF2PostLayout,
+)
+from qiskit.transpiler.passes.layout.vf2_layout import VF2LayoutStopReason
+from qiskit.transpiler.preset_passmanagers import common
+
+from mqt.predictor.rl.parsing import (
+    PreProcessTKETRoutingAfterQiskitLayout,
+    get_bqskit_native_gates,
+)
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+    from typing import Any
+
+    from bqskit import Circuit
+    from pytket._tket.passes import BasePass as tket_BasePass
+    from qiskit.transpiler.basepasses import BasePass as qiskit_BasePass
+
+
+class CompilationOrigin(str, Enum):
+    """Enumeration of the origin of the compilation action."""
+
+    QISKIT = "qiskit"
+    TKET = "tket"
+    BQSKIT = "bqskit"
+    GENERAL = "general"
+
+
+class PassType(str, Enum):
+    """Enumeration of the type of compilation pass."""
+
+    OPT = "optimization"
+    SYNTHESIS = "synthesis"
+    MAPPING = "mapping"
+    LAYOUT = "layout"
+    ROUTING = "routing"
+    FINAL_OPT = "final_optimization"
+    TERMINATE = "terminate"
+
+
+@dataclass
+class Action:
+    """Base class for all actions in the reinforcement learning environment."""
+
+    name: str
+    origin: CompilationOrigin
+    pass_type: PassType
+    transpile_pass: (
+        list[qiskit_BasePass | tket_BasePass]
+        | Callable[..., list[qiskit_BasePass | tket_BasePass]]
+        | Callable[
+            ...,
+            Callable[..., tuple[Any, ...] | Circuit],
+        ]
+    )
+
+
+@dataclass
+class DeviceIndependentAction(Action):
+    """Action that represents a static compilation pass that can be applied directly."""
+
+
+@dataclass
+class DeviceDependentAction(Action):
+    """Action that represents a device-specific compilation pass that can be applied to a specific device."""
+
+    transpile_pass: (
+        Callable[..., list[qiskit_BasePass | tket_BasePass]]
+        | Callable[
+            ...,
+            Callable[..., tuple[Any, ...] | Circuit],
+        ]
+    )
+    do_while: Callable[[dict[str, Circuit]], bool] | None = None
+
+
+# Registry of actions
+_ACTIONS: dict[str, Action] = {}
```

Comment:
Could we maybe handle this similarly to the registries in MQT Bench? Including the nice decorator solution for registering further passes.